WeatherMonitoringAndReportingService/Bots/RainBot.cs

```diff
@@ -0,0 +1,31 @@
+﻿using WeatherMonitoringAndReportingService.Config;
+using WeatherMonitoringAndReportingService.WeatherDetails;
+
+namespace WeatherMonitoringAndReportingService.Bots;
+
+public class RainBot:IBot
+{
+    private WeatherConfigurationService _weatherConfigurationService;
+
+    public RainBot(WeatherConfigurationService weatherConfigurationService)
+    {
+        _weatherConfigurationService = weatherConfigurationService;
+    }
```

Comment:
Replace with an interface called IWeatherConfigurationService that implements the GetBotConfiguration and UpdateBotConfiguration methods for it to be injected into the bot classes.